pkg/cmd/diagnose.go

```diff
@@ -44,60 +87,135 @@ func (o *DiagnoseOptions) Run() error {
 	}
 
 	log.Logger().Infof("Running in namespace: %s", util.ColorInfo(ns))
-	if o.showOption("version") {
-		err := printStatus(o, "Jenkins X Version", "jx", "version", "--no-version-check")
-		if err != nil {
-			return err
-		}
+	_, table := o.GetPackageVersions(ns, o.HelmTLS)
+
+	// os version
+	osVersion, err := o.GetOsVersion()
+	if err != nil {
+		log.Logger().Warnf("Failed to get OS version: %s", err)
+	} else {
+		table.AddRow("Operating System", util.ColorInfo(osVersion))
 	}
 
-	if o.showOption("status") {
-		err := printStatus(o, "Jenkins X Status", "jx", "status")
-		if err != nil {
-			return err
+	table.Render()
+
+	switch len(o.Args) {
+	//ToDo: This wont be required after the deprecated --show flag is removed
```

Comment:
Maybe moving all the code which is temporary or can be later removed into a dedicated function? WDYT?

Reply:
makes sense, will do :+1: